identity_iota_core/src/rebased/client/full_client.rs

```diff
@@ -253,18 +259,49 @@ impl<S> IdentityClient<S> {
     Ok(budget)
   }
 
-  async fn get_coin_for_transaction(&self) -> Result<iota_sdk::rpc_types::Coin, Error> {
-    let coins = self
-      .coin_read_api()
-      .get_coins(self.sender_address(), None, None, None)
-      .await
-      .map_err(|err| Error::GasIssue(format!("could not get coins; {err}")))?;
+  async fn get_coin_for_transaction(&self) -> Result<Coin, Error> {
+    const LIMIT: usize = 10;
+    let mut cursor = None;
+
+    let mut best_coin: Option<Coin> = None;
+
+    loop {
+      let coins = self
+        .coin_read_api()
+        .get_coins(self.sender_address(), None, cursor, Some(LIMIT))
+        .await?;
+
+      if coins.data.is_empty() {
+        return if let Some(coin) = best_coin {
+          Ok(coin)
+        } else {
+          Err(Error::GasIssue("no coins found for address".to_string()))
```

Comment:
Could you add the address to the error message? This might be helpful in case multiple accounts are used. ^^